crates/noirc_evaluator/src/ssa/acir_gen/operations/binary.rs

```diff
@@ -145,17 +145,24 @@ pub(crate) fn evaluate(
                 let l_c = var_cache.get_or_compute_internal_var_unwrap(binary.lhs, evaluator, ctx);
                 let mut r_c = var_cache.get_or_compute_internal_var_unwrap(binary.rhs, evaluator, ctx);
                 let predicate = get_predicate(var_cache,binary, evaluator, ctx).expression().clone();
-                //TODO avoid creating witnesses here.
-                let x_witness = r_c.get_or_compute_witness(evaluator, false).expect("unexpected constant expression"); 
-
-                let inverse = Expression::from(&constraints::evaluate_inverse(
-                    x_witness, &predicate, evaluator,
-                ));
-                InternalVar::from(constraints::mul_with_witness(
-                    evaluator,
-                    l_c.expression(),
-                    &inverse,
-                ))
+                if let Some(r_value) = r_c.to_const() {
+                    if r_value.is_zero() {
+                        panic!("Panic - division by zero");
```

Comment:
For reference, is this a panic that can happen because of user input or is this a panic that would indicate a compiler bug?

Reply:
It is because of a user input, you can refer to the PR description, it should be refactored using error handling.

Reply:
Hmm I was thinking of a way to indicate this for new panics we add while we do not have error handling, but perhaps its better this way as it forces us to look at all panics when error handling is added

Reply:
Ah sorry I didn't see the additional context